packages/vm/src/vm.ts

```diff
@@ -59,7 +59,9 @@ export class VM {
    * set to public due to implementation internals
    * @hidden
    */
-  public readonly _emit: (topic: string, data: any) => Promise<void>
+  public readonly _emit(topic: keyof VMEvents, data: any): Promise<void> {
```

Comment:
The linter is complaining about this being an invalid use of `readonly` now that you've included the implementation alongside the declaration.  I don't have an opinion about this in general but can we either remove `readonly` or else put the implementation back down below to make the linter happy?

Reply:
Oh yea readonly cannot be used in this spot in ts. Sorry was making the changes in github editor instead of real editor. Will pull the pr and make it good later today